python/src/scipp/compat/mantid.py

```diff
@@ -776,7 +775,8 @@ def convert_WorkspaceGroup_to_dataarray_dict(group_workspace, **kwargs):
     workspace_dict = {}
     for i in range(group_workspace.getNumberOfEntries()):
         workspace = group_workspace.getItem(i)
-        workspace_name = workspace.name().replace(group_workspace.name(), '')
+        workspace_name = workspace.name().replace(f'_{group_workspace.name()}',
```

Comment:
shouldn't this be `group_workspace.name()_`? I thought the name of the group came first?

Reply:
I thought so too, but at least in the case I tried it comes at the end.

Reply:
Hmm, actually there is no group name at all when grouping manually.

Reply:
Pushed an update that should handle both cases. Minor caveat: May strip more than necessary, leading to name clashes in the worst case, if someone uses `X_` and `X__` as workspaces names.